test/integration/replicaset/replicaset_test.go

```diff
@@ -162,19 +173,236 @@ func rmSimpleSetup(t *testing.T) (*httptest.Server, framework.CloseFunc, clients
+	}
+	return pod
+}
+
+func updatePodStatus(t *testing.T, podClient typedv1.PodInterface, pod *v1.Pod, updateStatusFunc func(*v1.Pod)) {
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		newPod, err := podClient.Get(pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		updateStatusFunc(newPod)
+		_, err = podClient.UpdateStatus(newPod)
+		return err
+	}); err != nil {
+		t.Fatalf("Failed to update status of pod %s: %v", pod.Name, err)
+	}
+}
+
+func getPods(t *testing.T, podClient typedv1.PodInterface, labelMap map[string]string) *v1.PodList {
+	podSelector := labels.Set(labelMap).AsSelector()
+	options := metav1.ListOptions{LabelSelector: podSelector.String()}
+	pods, err := podClient.List(options)
+	if err != nil {
+		t.Fatalf("Failed obtaining a list of pods that match the pod labels %v: %v", labelMap, err)
+	}
+	return pods
+}
+
+func updateRS(t *testing.T, rsClient typedv1beta1.ReplicaSetInterface, rsName string, updateFunc func(*v1beta1.ReplicaSet)) *v1beta1.ReplicaSet {
+	var rs *v1beta1.ReplicaSet
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		newRS, err := rsClient.Get(rsName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		updateFunc(newRS)
+		rs, err = rsClient.Update(newRS)
+		return err
+	}); err != nil {
+		t.Fatalf("Failed to update rs %s: %v", rsName, err)
+	}
+	return rs
+}
+
+// Verify ControllerRef of a RS pod that has incorrect attributes is automatically patched by the RS
+func testPodControllerRefPatch(t *testing.T, c clientset.Interface, pod *v1.Pod, ownerReference *metav1.OwnerReference, rs *v1beta1.ReplicaSet, expectedOwnerReferenceNum int) {
+	ns := rs.Namespace
+	podClient := c.Core().Pods(ns)
+	updatePod(t, podClient, pod.Name, func(pod *v1.Pod) {
+		pod.OwnerReferences = []metav1.OwnerReference{*ownerReference}
+	})
+
+	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		newPod, err := podClient.Get(pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return metav1.GetControllerOf(newPod) != nil, nil
+	}); err != nil {
+		t.Fatalf("Failed to verify ControllerRef for the pod %s is not nil: %v", pod.Name, err)
+	}
+
+	newPod, err := podClient.Get(pod.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to obtain pod %s: %v", pod.Name, err)
+	}
+	controllerRef := metav1.GetControllerOf(newPod)
+	if controllerRef.UID != rs.UID {
+		t.Fatalf("RS owner of the pod %s has a different UID: Expected %v, got %v", newPod.Name, rs.UID, controllerRef.UID)
+	}
+	ownerReferenceNum := len(newPod.GetOwnerReferences())
+	if ownerReferenceNum != expectedOwnerReferenceNum {
+		t.Fatalf("Unexpected number of owner references for pod %s: Expected %d, got %d", newPod.Name, expectedOwnerReferenceNum, ownerReferenceNum)
+	}
+}
+
+func setPodsStatus(t *testing.T, clientSet clientset.Interface, pods *v1.PodList, conditionStatus v1.ConditionStatus, lastTransitionTime time.Time) {
```

Comment:
Sure, i will change the name to `setPodsReadyCondition`! I decided to put `Status` as part of the name because the function also sets pods to be running: `pod.Status.Phase = v1.PodRunning`.